### DISCUSSION TITLE
How to store only latest key-value pairs on Log file?

### DISCUSSION THREAD

**Discussion by rottensunday** [microsoft/FASTER#852] [Q&A]
Hello. I want to use FASTER for caching data about some sets of objects. Features I'm interested in are:
1. Fast caching both in RAM and disk if data doesn't fit ram
2. Data persistance. If I need to restart my application, I want to have access to all this data without spending time loading it from external sources

These objects are fairly often updated, consider scenario: there are 10 million entries (big objects) in cache and every so often some of these objects are updated. To store new versions in cache I'm doing Upserts and TakeFullCheckpoints. But new versions are appended to .log files instead of changing it, so if I update 5 million entries 5 times each, I'll get 25 million new entries stored in .log file. Retrieving historical entries is not something I'm interesed in, so I wonder how can I store only most recent values for objects on disk? Otherwise I'll quickly go out of space on disk.

Thanks for any help! (BTW this library is really great, but hard to get into for newbies :D)

**Comment by badrishc** [comment]
You need to use snapshot type of checkpoint. It's an option in the call to take checkpoint.
